src/toil/provisioners/gceProvisioner.py

```diff
@@ -0,0 +1,915 @@
+        #  - ex_create_multiple_nodes is limited to 1000 nodes
+        #    - use a different function
+        #    - or write a loop over the rest of this function, with 1K nodes max on each iteration
+
+        if not preemptable:
+            logger.info('Launching %s non-preemptable nodes', numNodes)
+        else:
+            logger.info('Launching %s preemptable nodes', numNodes)
+
+        disk = {}
+        disk['initializeParams'] = {
+            'sourceImage': bytes('https://www.googleapis.com/compute/v1/projects/coreos-cloud/global/images/coreos-stable-1576-4-0-v20171206'),
+            'diskSizeGb' : self.nodeStorage }
+        disk.update({'boot': True,
+             #'type': 'bytes('zones/us-central1-a/diskTypes/local-ssd'), #'PERSISTANT'
+        #     'mode': 'READ_WRITE',
+        #     'deviceName': clusterName,
+             'autoDelete': True })
+        #instancesLaunched = driver.ex_create_multiple_nodes(
+        instancesLaunched = self.ex_create_multiple_nodes(
+                                '', nodeType, imageType, numNodes,
+                                location=self.zone,
+                                ex_service_accounts=sa_scopes,
+                                ex_metadata=metadata,
+                                ex_disks_gce_struct = [disk],
+                                description=self.tags,
+                                ex_preemptible = preemptable
+                                )
+        self.instanceGroup.add_instances(instancesLaunched)
+        workersCreated = 0
+        for instance in instancesLaunched:
+            if not self._copySshKeys(instance.public_ips[0], self.keyName):
+                continue
+            if self.config and self.config.sseKey or botoExists:
+                if not self._waitForNode(instance.public_ips[0], 'toil_worker'):
+                    continue
+                retries = 0
+                while True:
+                    try:
+                        if self.config and self.config.sseKey:
+                            self._rsyncNode(instance.public_ips[0], [self.config.sseKey, ':' + self.config.sseKey],
+                                            applianceName='toil_worker')
+                        if botoExists:
+                            self._rsyncNode(instance.public_ips[0], [self.botoPath, ':' + nodeBotoPath],
+                                            applianceName='toil_worker')
+                        break
+                    except Exception as e:
+                        if retries == 5:
+                            log.error("Failed to rysnc keys to worker with ip" % instance.public_ips[0])
+                            log.error('Exception %s', e)
+                            continue # givup
+                        else:
+                            logger.debug("Rsync to new nodes failed, trying again")
+                            retries += 1
+                            time.sleep(20*retries)
```

Comment:
This whole loop will wait 8 minutes trying to rsync.  Is it necessary to wait that long?

Reply:
It waited forever before.